packages/object/addon/index.js

```diff
@@ -286,28 +287,37 @@ export const observes = decoratorWithRequiredParams((desc, params) => {
     desc && desc.descriptor && typeof desc.descriptor.value === 'function'
   );
 
-  // hackity hackity haaaaaack
-  desc.extras = [
-    {
-      kind: 'field',
-      placement: 'own',
-      key: '__EMBER_DECORATORS_FINISH_CHAINS__',
-      descriptor: {
-        enumerable: false,
-        writable: true,
-        configurable: true,
-      },
-      initializer() {
-        if (!(this instanceof EmberObject) && !CHAINS_FINISHED.has(this)) {
-          Ember.finishChains(Ember.meta(this));
-
-          CHAINS_FINISHED.set(this, true);
+  // hasChainsFinished will be set to true when the first initializer field
+  // is added during class evaluation, then back to false when the finishers
+  // run. This is not ideal, but shouldn't be an issue in the future when we
+  // have the ability to add arbitrary initializers that don't have to be fields
+  if (hasChainsFinished === false) {
+    hasChainsFinished = true;
+
+    // hackity hackity haaaaaack
```

Comment:
:+1:

Reply:
:notes: To the tune of [*Yakety Yak*](https://en.wikipedia.org/wiki/Yakety_Yak) no doubt...